runtime/src/main/cpp/Atomic.cpp

```diff
@@ -30,6 +36,20 @@ template <typename T> T compareAndSwapImpl(KRef thiz, T expectedValue, T newValu
     return compareAndSwap(location, expectedValue, newValue);
 }
 
+inline AtomicReferenceLayout* asAtomicReference(KRef thiz) {
+    return reinterpret_cast<AtomicReferenceLayout*>(thiz + 1);
+}
+
+inline void lock(KRef thiz) {
```

Comment:
It is not clear from the signature of this function that it is applicable only to AtomicReference.
It makes sense to change it name or parameter type (or even moving to `AtomicReferenceLayout` class).

Reply:
Yep.